service/src/test/java/de/adorsys/psd2/sandbox/xs2a/pis/PaymentInitiationWithScaSteps.java

```diff
@@ -119,17 +121,27 @@ public void tryToInitiatePayment(String paymentService, String iban, String paym
 
     Request<PaymentInitiationSctJson> request = new Request<>(payment, headers);
 
-    ResponseEntity<TppMessage403PIS[]> response = template.exchange(
+    ResponseEntity<KeyedTpp403Messages> response = template.exchange(
         paymentService + "/" + paymentProduct,
         HttpMethod.POST,
         request.toHttpEntity(),
-        TppMessage403PIS[].class);
+        KeyedTpp403Messages.class);
 
     assertTrue(response.getStatusCode().is4xxClientError());
 
     context.setActualResponse(response);
   }
 
+  // TODO can't parse to TppMessage403PIS[] because we get `{tppMessages: TppMessage403PIS[]}`
+  private static class KeyedTpp403Messages {
```

Comment:
Litte bit confused by the naming. When they are custom errors. Why not call it CustomTpp403Messages or sth similar?

Reply:
I named the helper class `KeyedXXX` because returning `{value: {result: 1}}` instead of `{result: 1}` is sometimes referred to as `keying` or `keyed` JSON.